website/addons/base/__init__.py

```diff
@@ -127,7 +127,29 @@ def __init__(self, short_name, full_name, owners, categories,
                 'user_settings_default.mako',
             )
 
+        # Provide the path the the node_settings template
         self.node_settings_template = node_settings_template
+        node_settings_template_name = node_settings_template or '{}_node_settings.mako'.format(self.short_name)
+        addon_node_settings_path = os.path.join(
+            template_path,
+            node_settings_template_name
+        )
+        if node_settings_template and os.path.exists(addon_node_settings_path):
+            # If NODE_SETTINGS_TEMPLATE is defined, use that path.
+            self.node_settings_template = os.path.join(
```

Comment:
So addon developers must either implement the interface in the default node/user settings file or give a full path like:

``` NODE_SETTINGS_PATH = 'addons/s3/templates/s3_node_settings.mako' ```
or
``` NODE_SETTINGS_PATH = 'templates/project/addons/citations_node_settings.mako' ```